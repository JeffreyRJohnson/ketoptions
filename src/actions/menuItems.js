import uuid from 'uuid';
import database from '../firebase/firebase';

// ADD_MENU_ITEM
export const addMenuItem = menuItem => ({
  type: 'ADD_MENU_ITEM',
  menuItem
});

export const startAddMenuItem = (menuItemData = {}) => {
  return dispatch => {
    const {
      calories = 0,
      carbs = 0,
      fat = 0,
      menu_item = '',
      protein = 0,
      restaurant_name = ''
    } = menuItemData;

    const menuItem = {
      calories,
      carbs,
      fat,
      menu_item,
      protein,
      restaurant_name
    };

    return database
      .ref('menuItems')
      .push(menuItem)
      .then(ref => {
        dispatch(
          addMenuItem({
            id: ref.key,
            ...menuItem
          })
        );
      });
  };
};

// REMOVE_MENU_ITEM
export const removeMenuItem = ({ id } = {}) => ({
  type: 'REMOVE_MENU_ITEM',
  id
});

// EDIT_MENU_ITEM
export const editMenuItem = (id, updates) => ({
  type: 'EDIT_MENU_ITEM',
  id,
  updates
});
