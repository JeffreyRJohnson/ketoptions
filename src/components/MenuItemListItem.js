import React from 'react';
import { Link } from 'react-router-dom';

const MenuItemListItem = ({
  id,
  menu_item,
  restaurant_name,
  calories,
  carbs,
  fat,
  protein
}) => (
  <div>
    {/* <h2>{restaurant_name}</h2> */}
    <Link to={`/edit/${id}`}>
      <h3>{menu_item}</h3>
    </Link>
    <p>
      calories - {calories} protein - {protein} fat - {fat} carbs - {carbs}
    </p>
  </div>
);

export default MenuItemListItem;
