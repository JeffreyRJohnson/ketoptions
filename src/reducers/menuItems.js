// MenuItems Reducer

const menuItemsReducerDefaultState = [];

export default (state = menuItemsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_MENU_ITEM':
      return [...state, action.menuItem];
    case 'REMOVE_MENU_ITEM':
      return state.filter(({ id }) => id !== action.id);
    case 'EDIT_MENU_ITEM':
      return state.map(menuItem => {
        if (menuItem.id === action.id) {
          return {
            ...menuItem,
            ...action.updates
          };
        } else {
          return menuItem;
        }
      });
    default:
      return state;
  }
};
