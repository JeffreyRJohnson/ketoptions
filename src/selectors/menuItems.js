// Get visible menuItems

export default (menuItems, { text }) => {
  return menuItems.filter(menuItem => {
    const textMatch = menuItem.restaurant_name
      .toLowerCase()
      .includes(text.toLowerCase());

    return textMatch;
  });
};
