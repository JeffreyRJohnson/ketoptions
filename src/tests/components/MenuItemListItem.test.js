import React from 'react';
import { shallow } from 'enzyme';
import MenuItemListItem from '../../components/MenuItemListItem';
import menuItems from '../fixtures/menuItems';

test('should render MenuItemListItem with menuItems', () => {
  const wrapper = shallow(<MenuItemListItem {...menuItems[0]} />);
  expect(wrapper).toMatchSnapshot();
});
