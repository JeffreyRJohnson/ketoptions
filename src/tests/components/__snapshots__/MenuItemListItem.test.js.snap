// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MenuItemListItem with menuItems 1`] = `
<div>
  <h2>
    Wendys
  </h2>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      chilli - medium
    </h3>
  </Link>
  <p>
    calories - 
    400
     protein - 
    30
     fat - 
    20
     carbs - 
    5
  </p>
</div>
`;
